opennebula/provider.go

```diff
@@ -49,16 +55,23 @@ func Provider() *schema.Provider {
 			"opennebula_virtual_machine":       resourceOpennebulaVirtualMachine(),
 			"opennebula_virtual_network":       resourceOpennebulaVirtualNetwork(),
 			"opennebula_virtual_machine_group": resourceOpennebulaVMGroup(),
+			"opennebula_service":               resourceOpennebulaService(),
+			"opennebula_service_template":      resourceOpennebulaServiceTemplate(),
 		},
 
 		ConfigureFunc: providerConfigure,
 	}
 }
 
 func providerConfigure(d *schema.ResourceData) (interface{}, error) {
-	client := goca.NewDefaultClient(goca.NewConfig(d.Get("username").(string),
+	one_client := goca.NewDefaultClient(goca.NewConfig(d.Get("username").(string),
 		d.Get("password").(string),
 		d.Get("endpoint").(string)))
 
-	return goca.NewController(client), nil
+	flow_client := goca.NewDefaultFlowClient(
```

Comment:
if the flow_endpoint is not set you do not need the flow client.

You should instantiate the flow client only if the Flow endpoint is set.